### PR TITLE
pkg/obfuscate: redis obfuscation var to remove arguments

### PR DESCRIPTION
**Pull request by hannahkm**
### What does this PR do?

This PR creates and implements a new obfuscation config variable, `apm_config.obfuscation.redis.remove_all_args`, and corresponding environment variable `DD_APM_OBFUSCATION_REDIS_REMOVE_ALL_ARGS`. This variable, when enabled, allows the user to obfuscate all arguments for any given Redis command to a single "?".

### Motivation

Certain Redis commands can reveal sensitive information (i.e. hash, password, etc) that should not be public or saved to the backend. This PR will allow the user to enable full obfuscation of Redis commands to hide all arguments instead of just certain ones.

### Describe how to test/QA your changes

Changes will be tested by unit tests. For manual testing, set environment variable `DD_APM_OBFUSCATION_REDIS_REMOVE_ALL_ARGS` and/or config variable `apm_config.obfuscation.redis.remove_all_args` to true. The change should be read correctly by the agent and reflected in logs. 

Run `curl localhost:5012/debug/vars` to dump the runtime config of the agent. This should contain the configured values in the env vars mentioned previously.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by git-thuerk-done — APPROVED** [review]
Release notes look good for docs!

**Review by Kaderinho — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by shared components.

File: pkg/obfuscate/redis.go [hunk 1]
```diff
@@ -245,6 +245,43 @@ func obfuscateRedisCmd(out *strings.Builder, cmd string, args ...string) {
...
+	if len(args) == 0 {
+		return out.String()
+	}
+
+	out.WriteByte(' ')
+	switch strings.ToUpper(cmd) {
+	case "BITFIELD":
+		out.WriteString("?")
+		for _, arg := range args {
+			if arg == "SET" || arg == "GET" || arg == "INCRBY" {
```
Comment (knusbaum) [inline pkg/obfuscate/redis.go:268]: Above, we checked `strings.ToUpper(cmd)`.
Could these command args also be lower-case?

**Review by knusbaum — CHANGES_REQUESTED** [review]
This looks really good.

I only have a couple comments.

File: pkg/obfuscate/redis.go [hunk 2]
```diff
@@ -264,13 +264,15 @@ func (*Obfuscator) RemoveAllRedisArgs(rediscmd string) string {
 	switch strings.ToUpper(cmd) {
 	case "BITFIELD":
 		out.WriteString("?")
-		for _, arg := range args {
+		for _, a := range args {
+			arg := strings.ToUpper(a)
 			if arg == "SET" || arg == "GET" || arg == "INCRBY" {
-				out.WriteString(strings.Join([]string{"", arg, "?"}, " "))
+				out.WriteString(strings.Join([]string{"", a, "?"}, " "))
```
Comment (knusbaum) [inline pkg/obfuscate/redis.go:270]: Good attention keeping the original `a` here. :+1:

**Review by knusbaum — APPROVED** [review]
Looks great!

**Pull request merged**